plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1440,4 +1447,59 @@ private static ColumnMetadata getPageSinkIdColumn(List<String> otherColumnNames)
         }
         return new ColumnMetadata(columnName, TRINO_PAGE_SINK_ID_COLUMN_TYPE);
     }
+
+    public RemoteIdentifiers getRemoteIdentifiers(Connection connection)
+    {
+        return new JdbcRemoteIdentifiers(connection);
+    }
+
+    private class JdbcRemoteIdentifiers
```

Comment:
I would extract it separate file. Non static inner classes are kind of complex.

Reply:
refactored